# ==============================================================================
# FIXING COMMIT DIFF
# ==============================================================================
# File: dom/media/platforms/ffmpeg/FFmpegVideoDecoder.cpp
# Commit: 2e638f8a11df
# Full Hash: 2e638f8a11df2d0d7424059abc8ae65db580cf87
# Author: Martin Stransky <stransky@redhat.com>
# Date: 2020-04-24 03:06:24
# Description:
#   Bug 1632059 [Wayland] Use wayland display directly for va-api, r=jhorak
#   
#   Differential Revision: https://phabricator.services.mozilla.com/D72122
# ==============================================================================

diff -r ddbd6938e3f3 -r 2e638f8a11df dom/media/platforms/ffmpeg/FFmpegVideoDecoder.cpp
--- a/dom/media/platforms/ffmpeg/FFmpegVideoDecoder.cpp	Thu Apr 23 18:34:21 2020 +0000
+++ b/dom/media/platforms/ffmpeg/FFmpegVideoDecoder.cpp	Thu Apr 23 16:06:54 2020 +0000
@@ -178,7 +178,12 @@
   AVHWDeviceContext* hwctx = (AVHWDeviceContext*)mVAAPIDeviceContext->data;
   AVVAAPIDeviceContext* vactx = (AVVAAPIDeviceContext*)hwctx->hwctx;
 
-  mDisplay = mLib->vaGetDisplayWl(widget::WaylandDisplayGet()->GetDisplay());
+  wl_display* display = widget::WaylandDisplayGetWLDisplay();
+  if (!display) {
+    FFMPEG_LOG("Can't get default wayland display.");
+    return false;
+  }
+  mDisplay = mLib->vaGetDisplayWl(display);
 
   hwctx->user_opaque = new VAAPIDisplayHolder(mLib, mDisplay);
   hwctx->free = VAAPIDisplayReleaseCallback;